packages/jest-runner/src/index.ts

```diff
@@ -44,6 +44,7 @@ namespace TestRunner {
 class TestRunner {
   private _globalConfig: Config.GlobalConfig;
   private _context: JestTestRunnerContext;
+  readonly isSerial?: boolean;
```

Comment:
The `jest-runner` class right now is the source of truth for all test runners. This means it needs to have an `isSerial` property for the test scheduler to use it.

In the future, I think TestRunner should become an interface implemented by JestTestRunner.

Reply:
Yeah, sounds good 👍

Reply:
I tried and failed to do the interface patch. The runner interface depends on jest-haste-map, which depends on jest/@types. So I can't put it in types. And it needs to be used by jest-runner, which is depended on by @jest/core, so I can't put it in @jest/core. I'm not sure where it should go without creating a cycle. Can I create a new package? How does that work?

Reply:
Can it be in `jest-runner`? And yeah, a new package is fine if it doesn't fit anywhere else, although lifting some types out (like I've done in #9747 and #9749) to break cycles might be enough.